micrometer-core/src/test/java/io/micrometer/core/instrument/binder/system/FileDescriptorMetricsTest.java

```diff
@@ -20,76 +20,55 @@
 import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
 import org.junit.jupiter.api.Test;
 
-import java.lang.management.ManagementFactory;
 import java.lang.management.OperatingSystemMXBean;
-import java.lang.reflect.InvocationTargetException;
 
 import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.api.Assumptions.assumeFalse;
 import static org.junit.jupiter.api.Assumptions.assumeTrue;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
 
 /**
- * File descriptor metrics.
+ * Tests for {@link FileDescriptorMetrics}
  *
  * @author Michael Weirauch
  */
 class FileDescriptorMetricsTest {
     private MeterRegistry registry = new SimpleMeterRegistry();
 
-    @Test
-    @SuppressWarnings("restriction")
-    void fileDescriptorMetricsRuntime() {
-        // currently only supporting HotSpot JVM
-        final OperatingSystemMXBean osBean = ManagementFactory.getOperatingSystemMXBean();
-        assumeTrue(osBean instanceof com.sun.management.UnixOperatingSystemMXBean);
```

Comment:
Can't use `instanceof` on a class that will only be there when the JVM implementation is HotSpot. It would throw an exception on a different JVM.
HotSpot and J9 and supported now. Setting up a build that runs on J9 might be a good idea.